ceph-setup/build/build

```diff
@@ -38,24 +38,46 @@ fi
 echo "Running submodule update ..."
 git submodule update --init
 
-echo "Running autogen.sh ..."
-./autogen.sh
-echo "Running configure ..."
-./configure \
-  --disable-option-checking \
-  '--prefix=/usr' \
-  '--sbindir=/sbin' \
-  '--localstatedir=/var' \
-  '--sysconfdir=/etc' \
-  '--with-debug' \
-  '--with-nss' \
-  '--with-radosgw' \
-  '--disable-static' \
-  '--without-lttng' \
-  'CFLAGS= -Wno-unused-parameter' \
-  'CXXFLAGS=  -Wno-unused-parameter' \
-  --cache-file=/dev/null \
-  --srcdir=.
+# Flavor Builds support
+
+#cryptopp is not default for CMake
+AUTOCONF_FLAVOR_FLAGS="--without-cryptopp"
+
+if [ "${FLAVOR}" == "notcmalloc" ]
+then
+    echo "Detected notcmalloc flavor: will use flag: --without-tcmalloc or -DALLOCATOR=libc"
+    AUTOCONF_FLAVOR_FLAGS="$AUTOCONF_FLAVOR_FLAGS --without-tcmalloc"
+    export CEPH_EXTRA_CMAKE_ARGS="$CEPH_EXTRA_CMAKE_ARGS -DALLOCATOR=libc"
+else
+    AUTOCONF_FLAVOR_FLAGS="$AUTOCONF_FLAVOR_FLAGS --with-tcmalloc"
+    export CEPH_EXTRA_CMAKE_ARGS="$CEPH_EXTRA_CMAKE_ARGS -DALLOCATOR=tcmalloc"
+fi
+
+# make-dist was renamed in conjuntion with cmake cutover
+if [ -x make-dist ]
```

Comment:
if `make-dist` is going to be called later, no need to do the `if/else` here. This can just check if `make-dist` does *not* exist so that it can just run `autogen.sh`